flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -426,7 +426,12 @@ public void run() {
 			try {
 				reporter.report();
 			} catch (Throwable t) {
-				LOG.warn("Error while reporting metrics", t);
+				if (LOG.isDebugEnabled()) {
+					LOG.debug("Error while reporting metrics {}", t);
+				}
+				else {
+					LOG.warn("Error while reporting metrics", t.getMessage());
```

Comment:
We could also keep it as a warning. I'm just wondering whether we should log the stack traces of an exception where we now that some of these exceptions are actually caused by us and are not critical. This will make people wary even though there is nothing to be concerned about.

Reply:
I'd rather be on the safe side and log the stack-trace in case it is not caused by us. Whether a report failing or not being critical depends on the context and user.

For example, a reporter failing consistently over a large period of time should show up in the logs in a manner that allows the user to make make a conscious decision whether the cluster has to be restarted or not. However in the worst case you get a `null` error message, which is entirely unhelpful for making said decision.

If we want to go for temporary band-aids then iterate over all reporters and catch `ConcurrentModificationExceptions`. Then you don't have these somewhat-expected intermittent exceptions caused by us anymore.

Reply:
Do we know where `ConcurrentModificationExceptions` can be thrown in the reporters @zentol?

Reply:
sure we do, when iterating over the internal maps `report()`. Some reporters (like Prometheus) already catch them and continue on.

Reply:
We could also think about only fixing the `Slf4jReporter` used in the e2e tests. But this is really only a band aid.

Reply:
I agree that it is a band-aid. That's why I haven't made that change yet, I wanted to solve this properly :(